via/resources.py

```diff
@@ -45,7 +51,18 @@ def _normalise_url(cls, url):
         Take a URL from user input (eg. a URL input field or path parameter) and
         convert it to a normalized form.
         """
-        if not (url.startswith("http://") or url.startswith("https://")):
-            url = "https://" + url
+
+        try:
+            parsed = urlparse(url)
+        except ValueError as exc:
+            raise BadURL(url) from exc
+
+        if not parsed.scheme:
+            if not parsed.netloc:
+                # Without a scheme urlparse often assumes the domain is the
+                # path. To prevent this add a fake scheme and try again
```

Comment:
I'm 100% sure there's a very good explanation for this behaviour on urlparse but this always surprises me.

Reply:
I think it's because it's logically impossible to tell the difference, so they had to choose:

```
# Totally cool!
http://example.com/path/localhost
http://localhost/example.com/path
http://path/localhost/example.com

# Erm...
example.com/path/localhost
localhost/example.com/path
path/localhost/example.com

# Um...
path/localhost
example.com/path
localhost/path

# Gah...
path
example.com
localhost
```